fdbclient/BackupAgentBase.actor.cpp

```diff
@@ -257,6 +258,24 @@ std::pair<Version, uint32_t> decodeBKMutationLogKey(Key key) {
 	    bigEndian32(*(int32_t*)(key.begin() + backupLogPrefixBytes + sizeof(UID) + sizeof(uint8_t) + sizeof(int64_t))));
 }
 
+bool isSystemKey(KeyRef key) {
+	return key.size() && key[0] == systemKeys.begin[0];
+}
+
+EncryptCipherDomainId getEncryptDomainId(MutationRef m) {
```

Comment:
`CommitProxyServer::getEncryptDetailsFromMutationRef` does very similar purpose, however, it takes `ProxyCommitData` as input, which can be easily avoided. Can we audit the method and maybe move the logic into one of the EncryptionUtil classes? 

Also, I noticed CommitProxy method does handle `SetVersionstampedKey` usecase, at the least, it is better to extend the method to do it.

Reply:
I think there's two different use cases here, for backup we are extracting the tenantID whereas in `getEncryptDetailsFromMutationRef` we are extracting the encryption domain id. I think it would be confusing if both used the same method

Reply:
I can modify the code to make sure `getEncryptDetailsFromMutationRef` respects the versionstamed case